### ISSUE TITLE
FreeSWITCH: Runtime issues (Aborted)

### ISSUE THREAD

**Issue by BlueMax** [openwrt/telephony#30]
Starting 'freeswitch' just echoes 'Aborted' (uClibc message) and the process dies. I've never seen Freeswitch in action but that probably is not correct. Arguments '-h' and '-version' do work though.

Edit: The two commits (https://github.com/openwrt/telephony/commit/94fd5f5e3504bbb0077ace2641e04cbe61ec66a7 https://github.com/openwrt/telephony/commit/163874a260ae313f9c6f29dd89a6897e83c4cb0c) fix building but do not lead to sane binaries. The same compiling issues are handled already in 'configure.ac' but they are currently limited to ARM hosts(?). I will test how that affects the build (as there are even more flags set).

```
case "$host" in
     arm*-linux-gnueabi*|arm*-*-linux-*)
[...]
apr_cv_tcp_nodelay_with_cork=yes
ac_cv_file_dbd_apr_dbd_mysql_c=no
[...]
```

Edit: Bingo! Seems to work. Will revoke the two commits and prepare some new...

**Comment by BlueMax** [comment]
The $hosts=arm* templates/overrides seem to be ok. Now its running (tested on x86).

**Comment by sageahn** [comment]
I got the upstream repo today(Apr 23, 2015) and still got the same "aborted" message from x86 and VirtualBox image environment. any comment please.

**Comment by BlueMax** [comment]
Mazi Lo (maintainer) took a different approach. I don't know why. You can try [this patch](http://pastebin.com/LkD9qY9t) (put in `feeds/telephony/net/freeswitch/patches`) that actually reflects my patches (and worked for me). Also create this patch (https://github.com/BlueMax/telephony/commit/5d07b2bc061cdcb3168806c7702255df7f7e980b) to fix [this bug](https://forum.openwrt.org/viewtopic.php?pid=263333#p263333). I raised pull requests for these but they were rejected.

**Comment by sageahn** [comment]
I am at c5e7c034e45e7bbf3f703f13f74b76de934c6725 in feeds/telephony repo. those two patches you mentioned are already applied.

**Comment by BlueMax** [comment]
The `sofia.xml` patch might have been patched upstream (idk) but the `configure.ac` is not included (https://github.com/openwrt/telephony/blob/master/net/freeswitch/patches/configure_ac.patch).

**Comment by sageahn** [comment]
Hi BlueMax,

I noticed that your patch and Mazi's patch are different. however. those patches do not fix this issue. still found the aborted message unfortunately.

**Comment by BlueMax** [comment]
Then i don't know, sorry. It worked for me at that time.

**Comment by skyblaster** [comment]
Just testing FreeSWITCH from the 15.05-rc1 package repo on a WNDR3700v4 and am getting the Aborted message right away. Not sure how to provide any useful information here since nothing is written to the logs.
I see the package revisions are the same between rc1, rc2 (just released today), and snapshot, so am guessing they're all broken....at least for the ar71xx target.

**Comment by skyblaster** [comment]
Just confirmed the same error with 15.05-rc2.

**Comment by micmac1** [comment]
Hello @BlueMax @sageahn @skyblaster 

freeswitch-stable got added to the tree as a drop-in replacement and it doesn't have this issue. Can you please try it and consider closing this issue?

Kind regards,
Sebastian

**Comment by skyblaster** [comment]
I've been running a snapshot build of freeswitch-stable v1.6.17 from a late May/early June on a RPi2 without a hitch.
Would love to see a cherry picked build added to [branch:lede-17.01](https://github.com/openwrt/telephony/tree/lede-17.01)

**Comment by jslachta** [comment]
Hi @skyblaster 

freeswitch-stable is not the same package as freeswitch, the [branch:lede-17.01](https://github.com/openwrt/telephony/tree/lede-17.01) should receive only security&bug fixes.

**Comment by skyblaster** [comment]
I'm well aware of the _freeswitch_ vs _freeswitch-stable_ packages. It's _freeswitch-stable_ that I've been running successfully since June 2017.
Due to #118, I don't think _freeswitch_ even builds. It may still suffer from the aborted message that this ticket covers??

Regarding [branch:lede-17.01](https://github.com/openwrt/telephony/tree/lede-17.01), the last commit was over a year ago. I see you and @jow were set to fix that with #132, but no actual backporting occurred. Should that ticket be reopened?

**Comment by skyblaster** [comment]
I don't know when FreeSWITCH 1.8 will be finalised, but perhaps @micmac1 could cherry pick a 1.6.19 build to backport to [branch:lede-17.01](https://github.com/openwrt/telephony/tree/lede-17.01) ? Then when 1.8 comes out you could start a new/separate package?
